faulty_code.py

```
@@ -0,0 +1,86 @@
+    def get_token(self):
+        """Bug: hardcoded secret, weak token generation."""
+        import hashlib
+        secret = "super_secret_key_123"
+        token = hashlib.md5(self.user.encode()).hexdigest()
+        return token
+
+
+def fetch_data(url):
+    """Fetch data. Bug: no timeout, no SSL verification, no error handling."""
+    import requests
+    response = requests.get(url, verify=False)
```

Ruff (0.15.13)

[error] Probable use of `requests` call without timeout

(S113)